universe/interface.go

```diff
@@ -1235,3 +1236,90 @@ type IgnoreTree interface {
 	QueryTuples(context.Context, asset.Specifier,
 		...IgnoreTuple) TupleQueryResp
 }
+
+// BurnLeaf is a type that represents a burn leaf within the universe tree.
+type BurnLeaf struct {
+	// UniverseKey is the key that the burn leaf is stored at.
+	UniverseKey LeafKey
```

Comment:
Since we're creating a new tree, should we make sure we don't introduce the same collision issue again? Meaning we should use a fixes leaf key (that also includes the asset ID in case of a grouped asset) here?
Not sure if that changes the overall semantics though or would cause us to not be able to re-use some of the existing code. Mostly just thinking aloud.